app/jobs/autotest_run_job.rb

```
@@ -17,6 +17,13 @@ def perform(host_with_port, role_id, assignment_id, group_ids, user: nil, collec
     unless user.nil?
       TestRunsChannel.broadcast_to(user, { status: 'completed', job_class: 'AutotestRunJob' })
     end
+  rescue ServiceUnavailableException => e
+    status.catch_exception(e)
+    status[:status] = 'service_unavailable'
+    unless user.nil?
+      TestRunsChannel.broadcast_to(user, { **status.to_h, job_class: 'AutotestRunJob' })
+    end
+    raise e
```

Right, so I don't actually think you need to re-raise this error. It's sufficient to just broadcast the status. This could be related to the trouble you were having with the test cases (and perhaps the change you made to the `AutotestJob` superclass, which could be reverted)

---

thank you, good point!